src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -72,14 +75,15 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
         require(msg.sender == parentVault, NotAuthorized());
         require(marketParams.loanToken == asset, LoanAssetMismatch());
 
-        if (shares[marketId] == 0 && assets > 0) allMarketParams.push(marketParams);
+        if (assets > 0) IMorpho(morpho).supply(marketParams, assets, 0, address(this), hex"");
+        // TODO: justify safe casting
+        int256 change =
+            int256(IMorpho(morpho).expectedSupplyAssets(marketParams, address(this))) - int256(allocation(marketParams));
 
-        if (assets > 0) {
-            (, uint256 mintedShares) = IMorpho(morpho).supply(marketParams, assets, 0, address(this), hex"");
-            shares[marketId] += mintedShares;
+        if (indexInListCtz[marketId] == 0 && IMorpho(morpho).supplyShares(marketId, address(this)) > 0) {
```

Comment:
The purpose of ctz / indexInlistCtz is not clear. Why not just store the index? Note that having `index = 0` for absent markets is fine as long as you don't read its position unless you know it's in the array (if you can't know that you can just store index+1).

Reply:
Index+1 is way better indeed